hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/DeletedDirectoryTableHandler.java

```diff
@@ -0,0 +1,165 @@
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manages records in the Deleted Directory Table, updating counts and sizes of
+ * pending Directory Deletions in the backend.
+ */
+public class DeletedDirectoryTableHandler implements  OmTableHandler {
+
+  private ReconNamespaceSummaryManagerImpl reconNamespaceSummaryManager;
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(DeletedTableHandler.class);
+
+  public DeletedDirectoryTableHandler(
+      ReconNamespaceSummaryManagerImpl reconNamespaceSummaryManager) {
+    this.reconNamespaceSummaryManager = reconNamespaceSummaryManager;
+  }
+
+  /**
+   * Invoked by the process method to add information on those directories that
+   * have been backlogged in the backend for deletion.
+   */
+  @Override
+  public void handlePutEvent(OMDBUpdateEvent<String, Object> event,
+                             String tableName,
+                             Collection<String> sizeRelatedTables,
+                             HashMap<String, Long> objectCountMap,
+                             HashMap<String, Long> unreplicatedSizeCountMap,
+                             HashMap<String, Long> replicatedSizeCountMap)
+      throws IOException {
+    String countKey = getTableCountKeyFromTable(tableName);
+    String unReplicatedSizeKey = getUnReplicatedSizeKeyFromTable(tableName);
+
+    if (event.getValue() != null) {
+      OmKeyInfo omKeyInfo = (OmKeyInfo) event.getValue();
+      objectCountMap.computeIfPresent(countKey, (k, count) -> count + 1L);
+      Long newDeletedDirectorySize =
+          fetchSizeForDeletedDirectory(omKeyInfo.getObjectID());
+      unreplicatedSizeCountMap.computeIfPresent(unReplicatedSizeKey,
+          (k, size) -> size + newDeletedDirectorySize);
+    } else {
+      LOG.warn("Put event does not have the Key Info for {}.",
+          event.getKey());
+    }
+  }
+
+  /**
+   * Invoked by the process method to remove information on those directories
+   * that have been successfully deleted from the backend.
+   */
+  @Override
+  public void handleDeleteEvent(OMDBUpdateEvent<String, Object> event,
+                                String tableName,
+                                Collection<String> sizeRelatedTables,
+                                HashMap<String, Long> objectCountMap,
+                                HashMap<String, Long> unreplicatedSizeCountMap,
+                                HashMap<String, Long> replicatedSizeCountMap)
+      throws IOException {
+    String countKey = getTableCountKeyFromTable(tableName);
+    String unReplicatedSizeKey = getUnReplicatedSizeKeyFromTable(tableName);
+
+    if (event.getValue() != null) {
+      OmKeyInfo omKeyInfo = (OmKeyInfo) event.getValue();
+      objectCountMap.computeIfPresent(countKey, (k, count) -> count - 1L);
+      Long newDeletedDirectorySize =
+          fetchSizeForDeletedDirectory(omKeyInfo.getObjectID());
```

Comment:
You are right @devmadhuu, we can address that in a separate Jira issue. It will require its own design discussions and testing, and including it in this pull request would make it overly complex since this PR is already quite big. Currently, I am in the process of creating a small document outlining how we can make this service snapshot-aware, and I will share it with here shortly.